pages/common/views/components/datatable.jsx

```diff
@@ -1,48 +1,109 @@
-import React from 'react';
-import { map, isEmpty, isUndefined } from 'lodash';
+import React, { Component, Fragment } from 'react';
+import classnames from 'classnames';
+import { map, isEmpty, isUndefined, size } from 'lodash';
 import { getValue } from '../../../../lib/utils';
 import DatatableHeader from '../containers/datatable-header';
 
-const Table = ({
-  data,
-  schema,
-  formatters,
-  sortable
-} = {}) => {
-  if (isUndefined(data)) {
-    throw new Error('data must be provided');
+class Table extends Component {
+  constructor(options) {
+    super(options);
+    this.toggleContent = this.toggleContent.bind(this);
+    this.isExpandable = this.isExpandable.bind(this);
   }
-  const columns = !isEmpty(schema)
-    ? schema
-    : Object.keys(data[0]).reduce((obj, key) => ({ ...obj, [key]: {} }), {});
-  return (
-    <table className="govuk-react-datatable">
-      <thead>
-        <tr>
+
+  componentDidMount() {
+    if (!this.props.ExpandableRow) {
+      return;
+    }
+    const expanded = this.props.data.reduce((obj, { id }) => {
+      return { ...obj, [id]: false };
+    }, {});
+    this.setState({ expanded });
+  }
+
+  toggleContent(id) {
+    const { expanded } = this.state;
+    expanded[id] = !expanded[id];
+    this.setState({ expanded });
+  }
+
+  isExpandable(id) {
+    if (this.props.ExpandableRow) {
+      if (!this.state) {
+        return true;
+      }
+      return this.state && this.state.expanded[id];
+    }
+  }
+
+  render() {
+    const {
+      data,
+      schema,
+      sortable,
+      ExpandableRow
+    } = this.props;
+    if (isUndefined(data)) {
+      throw new Error('data must be provided');
+    }
+    const columns = !isEmpty(schema)
+      ? schema
+      : Object.keys(data[0]).reduce((obj, key) => ({ ...obj, [key]: {} }), {});
+    return (
+      <table className="govuk-react-datatable">
+        <thead>
+          <tr>
+            {
+              map(columns, (column, key) =>
+                <DatatableHeader key={key} id={key} sortable={sortable} { ...column } />
+              )
+            }
+          </tr>
+          <tr id="filter-header"></tr>
+        </thead>
+        <tbody>
           {
-            map(columns, (column, key) =>
-              <DatatableHeader key={key} id={key} sortable={sortable} { ...column } />
-            )
+            data.map(row => (
+              <Fragment key={row.id}>
+                <tr
+                  onClick={ExpandableRow && (() => this.toggleContent(row.id))}
+                  className={classnames({
+                    expandable: ExpandableRow,
+                    restrictions: row.notes,
```

Comment:
This probably wants removing as it's specific to this one implementation.

Perhaps just do it as a formatter on the `name` column?

As an aside, for accessibility reasons we might want to have the "there are restrictions" indicator be actual DOM rather than css pseudocontent.